oresat_c3/services/edl.py

```diff
@@ -320,170 +339,360 @@ def _run_cmd(self, request: EdlCommandRequest) -> EdlCommandResponse:
+
+
+class EdlFileReciever(CfdpUserBase):
+    """CFDP receiver for file uploads."""
 
     def __init__(self, upload_dir: str, fwrite_cache: OreSatFileCache):
-        self.upload_dir = upload_dir
-        Path(upload_dir).mkdir(parents=True, exist_ok=True)
-        self.fwrite_cache = fwrite_cache
-        self.file_name = ""
-        self.file_data = b""
-        self.file_data_len = 0
-        self.f = None
-        self.offset = 0
-        self.last_nak = 0
-        self.last_indication = Indication.NONE
-        self.last_pdu_ts = 0.0
-        self.send_fin_ts = 0.0
-        self.checksum_matched = False
-
-    def reset(self):
-        """Reset the EDL file receiver."""
-
-        logger.info("reset")
-        if self.f is not None:
-            self.f.close()
-            self.f = None
-        self.offset = 0
-        self.last_nak = 0
-        self.file_name = ""
-        self.file_data = b""
-        self.file_data_len = 0
-        self.last_indication = Indication.NONE
-        self.checksum_matched = False
-
-    def _init_recv(self, pdu: MetadataPdu):
-        if self.f is None:
-            self.reset()
-            self.file_name = pdu.dest_file_name
-            self.file_data_len = pdu.file_size
-            logger.info(f"{self.file_name} {self.file_data_len} started")
-            self.f = open(f"{self.upload_dir}/{self.file_name}", "wb")  # type: ignore
-
-    def _recv_data(self, pdu: FileDataPdu) -> Union[NakPdu, None]:
-        if pdu.offset != self.offset:
-            return self._make_nak(pdu)
-
-        self.f.write(pdu.file_data)  # type: ignore
-        self.offset += len(pdu.file_data)
-        self.file_data += pdu.file_data
-        if self.offset >= self.file_data_len:
-            logger.info("file transfer done")
-
-        logger.info(f"{self.last_indication.name} {self.offset} write")
-        return None
-
-    def _eof(self, pdu: EofPdu) -> AckPdu:
-        condition_code = ConditionCode.NO_ERROR
-        checksum = zlib.crc32(self.file_data).to_bytes(4, "little")
-        if checksum == pdu.file_checksum:
-            condition_code = ConditionCode.FILE_CHECKSUM_FAILURE
-            logger.info("file checksum matched")
-            self.checksum_matched = True
-        else:
-            logger.info("file checksum does not match")
+        path = Path(upload_dir)
+        path.mkdir(parents=True, exist_ok=True)
+        super().__init__(vfs=PrefixedFilestore(path))
 
-        if self.f is not None and self.checksum_matched:
-            self.f.close()
-            self.f = None
-            try:
-                self.fwrite_cache.add(f"{self.upload_dir}/{self.file_name}", consume=True)
-                logger.info(f"file {self.file_name} moved to fwrite cache")
-            except (ValueError, FileNotFoundError) as e:
-                logger.error(e)
-            logger.info(f"{self.file_name} {self.file_data_len} ended")
+        SOURCE_ID = ByteFieldU8(0)
+        DEST_ID = ByteFieldU8(1)
```

Comment:
Is there a way we can get this as a configurable field from oresat-configs?

Reason I ask is because there may be more than one destination in the ground-segment. and it's probably a bad idea for them to all the share the ID 1.

Reply:
This is a good idea, the IDs here were picked solely as placeholders. Realistically the whole MIB, the Management Information Base, should probably live in oresat-configs. The CFDP spec is pretty scarce on how the MIB should be implemented and synchronized, a very "exercise left for the reader" vibe. For our purposes oresat-configs is probably the best place for that to live.

I do however think that the implementation is beyond the scope of this PR. Once the MIB is implemented in oresat-configs, then a new PR here can utilize it.

Reply:
Fair enough- maybe I can take that on actually.

I just want to make sure that initial testability of any PR is good enough that we can try and do a basic AI&T run.

For yamcs- I suppose we could have everything share one bucket in just one instance of yamcs for now but we should not let this settle.

Reply:
If you could make an issue outlining an "improved configurability" story so we don't lose track of this.

Once it's done- feel free to mark this as resolved.